importlib_metadata/__init__.py

```diff
@@ -548,7 +550,8 @@ def locate_file(self, path):
         """
 
     @classmethod
-    def from_name(cls, name):
+    @validate_arguments
+    def from_name(cls, name: Annotated[str, Field(min_length=1)]):
```

Comment:
Why are we introducing a dependency here? Won't we have to port this to CPython?

Also, personally, I'm unconvinced in introducing a dependency for such a simple check :sweat_smile:

Reply:
Over a year ago, I presented at the Python Language Summit a [proposal to use annotations](https://github.com/jaraco/transformers/blob/main/Annotations%20as%20transforms.md) to provide validators and transformers. The recommendation then was for me to explore pydantic, so I'm doing that now. I'm not certain I want to add a dependency, but I also want to explore the space. If it works well, I'll want to expand its usage to other similar patterns. As for porting to CPython, it's my understanding that parts of pydantic are being ported to CPython. I'm not sure if these parts are or not, but if not, this and related use-cases can help inform a rationale why they should.

Reply:
As much as I'd like to explore the use of pydantic, I'm now realizing that the cost of adding a dependency here is very high, not only because of the CPython integration but also because of the race condition with setuptools_scm. I'm going to abandon this approach in favor of a manual check in #391. I would eventually like to explore this concept again, but today is not the day.